src/trusted/service_runtime/nacl_syscall_common.c

```diff
@@ -1527,12 +1527,13 @@ static int32_t MunmapInternal(struct NaClApp *nap, uintptr_t sysaddr, size_t len
    * zero-filled pages, which should be copy-on-write and thus
    * relatively cheap.  Do not open up an address space hole.
    */
-  if (MAP_FAILED == mmap((void *) sysaddr,
-                         length,
-                         PROT_NONE,
-                         MAP_PRIVATE | MAP_ANONYMOUS | MAP_FIXED,
-                         -1,
-                         (off_t) 0)) {
+  if (MAP_FAILED == lind_mmap((void *) sysaddr,
```

Comment:
So we're still mmaping zeros here again to munmap, but you added munmap in SafePOSIX. I'm a bit confused as to what the goal is still.

Reply:
We are still mmapping 0s, I added munmap to SafePOSIX to complete the API, because I think we'd really have to futz around with NaCl's internal memory map in order to change this behavior and not have lots of seg faults or a huge memory leak.

Reply:
Yeah thats fine. We should continue to do what they implemented, I just want to make sure we don't get confused about this in the future.